server/src/main/java/org/opensearch/common/blobstore/VerifyingMultiStreamBlobContainer.java

```diff
@@ -31,4 +35,27 @@ public interface VerifyingMultiStreamBlobContainer extends BlobContainer {
      * @throws IOException if any of the input streams could not be read, or the target blob could not be written to
      */
     void asyncBlobUpload(WriteContext writeContext, ActionListener<Void> completionListener) throws IOException;
+
+    /**
+     * Creates an async callback of a {@link ReadContext} containing the multipart streams for a specified blob within the container.
+     * @param blobName The name of the blob for which the {@link ReadContext} needs to be fetched.
+     * @param listener  Async listener for {@link ReadContext} object which serves the input streams and other metadata for the blob
+     *
+     * @opensearch.experimental
+     */
+    void readBlobAsync(String blobName, ActionListener<ReadContext> listener);
+
+    /**
+     * Asynchronously downloads the blob to the specified location using an executor from the thread pool.
+     * @param blobName The name of the blob for which needs to be downloaded.
+     * @param fileLocation The path on local disk where the blob needs to be downloaded.
+     * @param threadPool The threadpool instance which will provide the executor for performing a multipart download.
+     * @param completionListener Listener which will be notified when the download is complete.
+     *
+     * @opensearch.experimental
+     */
+    default void asyncBlobDownload(String blobName, Path fileLocation, ThreadPool threadPool, ActionListener<String> completionListener) {
```

Comment:
Is this ever intended to be overriden? If no, then can we make this a static function in a helper class?

Reply:
Depending on the ThreadPool to be used, this might be overridden in `S3BlobContainer`.
Its still in flux and I think we should move ahead with this for now. 
I can finalize this with the S3 changes.